README.md

```diff
@@ -1235,6 +1237,20 @@ When a validator implements this interface, the method is called with the field'
 
 See [schema.IP](https://godoc.org/github.com/rs/rest-layer/schema#IP) validator for an implementation example.
 
+## JSON-Patch
+
+When patching deeply nested documents, it is more convenient to use protocol designed especially for this, instead
+of top level document field replacement as described in [RFC-5789](http://tools.ietf.org/html/rfc5789). For this purpose `rest-layer` implements [JSON-Patch/RFC-6902](https://tools.ietf.org/html/rfc6902). In addition JSON-Patch allows field deletion, which is not possible with the simple field replacement.
+
+Consider using JSON-Patch always with `If-Match`, to be sure which document is actually patched. In that case the updated document is already known, so HTTP responce body may be omitted via: `Prefer: return=minimal` HTTP requres header.
```

Comment:
minor typo: requres => request

The `If-Match` concurrency control is already mentioned elsewhere https://github.com/rs/rest-layer#data-integrity-and-concurrency-control. We should consider how much of that we want to repeat. A suggestion:

> If using `If-Match` concurrency control as described in the [data control and integrity section]( https://github.com/rs/rest-layer#data-integrity-and-concurrency-control), you could potentially choose to calculate the body of new object client side. Note that the response body for a successful operation can be omitted by supplying a HTTP request header: `Prefer: return=minimal`.

That said, I suppose this header can also be used for other requests? At least POST, PUT and PATCH without JSON-Patch. Is there some common place we could mention this in the docs?

Other than that, field selection works like for any other request (GET, POST, PUT), and could be used for limiting the response size. I don't know if it's worth mentioning though.

Reply:
I wanted to highlight what is the intended use for patching with JSON-Patch, alongside `If-Match`. I don't recall any common place to describe common HTTP methods functionality. I will see where I can put this.